Utilities/XrdAdaptor/src/XrdRequestManager.cc

```diff
@@ -118,6 +121,50 @@ static void SendMonitoringInfo(XrdCl::File &file) {
   }
 }
 
+std::string *GetQueryTransport(const XrdCl::URL &url, uint16_t query) {
+  XrdCl::AnyObject result;
+  XrdCl::DefaultEnv::GetPostMaster()->QueryTransport(url, query, result);
+  std::string *method;
+  result.Get(method);
+  return method;
+}
+
+static void TracerouteRedirections(const XrdCl::HostList *hostList) {
+  int idx_redirection = 1;
+  std::string traceroute_str = "-------------------------------\nTraceroute:\n";
+  for (auto itr = hostList->begin(); itr != hostList->end(); ++itr) {
+    // Query info
+    std::string *stack_ip_method = GetQueryTransport(itr->url, XrdCl::StreamQuery::IpStack);
+    std::string *ip_method = GetQueryTransport(itr->url, XrdCl::StreamQuery::IpAddr);
+    std::string *auth_method = GetQueryTransport(itr->url, XrdCl::TransportQuery::Auth);
+    std::string *hostname_method = GetQueryTransport(itr->url, XrdCl::StreamQuery::HostName);
+
+    // Organize redirection info
+    std::string redirection = std::to_string(idx_redirection) + ". || ";
+    redirection = redirection + *hostname_method + " / " + *stack_ip_method + " / " + *ip_method + " / " +
+                  std::to_string(itr->url.GetPort());
+    if (!auth_method->empty()) {
+      redirection = redirection + " / " + *auth_method;
+    } else {
+      redirection = redirection + " / " + "no auth";
+    };
+    if (itr->loadBalancer == 1) {
+      redirection = redirection + " / load balancer";
+    } else {
+      redirection = redirection + " / " + "endpoint";
+    };
+    redirection = redirection + " ||" + " \n";
+    traceroute_str = traceroute_str + redirection;
+    delete stack_ip_method;
+    delete ip_method;
+    delete auth_method;
+    delete hostname_method;
+    ++idx_redirection;
+  }
+  traceroute_str = traceroute_str + "-------------------------------";
+  edm::LogInfo("XrdAdaptorLvl2") << traceroute_str;
```

Comment:
Since the whole function is only about formatting an INFO-level message, I'd suggest to reorganize it along
```cpp
  edm::LogInfo("XrdAdaptorLvl2").log([hostList](auto& li) {
    int idx_redirection = 1;
    li << "-------------------------------\nTraceroute:\n";
    ...
  });
```
to avoid crafting the message when not needed. (my earlier suggestion on the use of `fmt::format()` could be combined with this suggestion as `li.format(<same arguments>);`)